ncrypt/smime.c

```diff
@@ -2524,25 +2533,36 @@ int smime_class_send_menu(struct Email *e)
                           _("123")))
               {
                 case 1:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes128");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes128", &errmsg);
                   break;
                 case 2:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes192");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes192", &errmsg);
                   break;
                 case 3:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes256");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes256", &errmsg);
                   break;
               }
               break;
 
             case 4:
-              FREE(&C_SmimeEncryptWith);
+              C_SmimeEncryptWith = NULL;
             /* (c)lear */
             /* fallthrough */
             case -1: /* Ctrl-G or Enter */
               choice = 0;
               break;
           }
+          if (CSR_RESULT(rc) != CSR_SUCCESS)
+          {
+            if (!mutt_buffer_is_empty(&errmsg))
+            {
+              mutt_error("%s", errmsg.data);
+              FREE(&errmsg.data);
+            }
+          }
```

Comment:
Please change the `FREE()` to a `mutt_buffer_dealloc()`.
Also, move it after the `if` clause, so it's at the same scope as the `mutt_buffer_make()`.

Currently, the `Buffer` will allocate a string in `Buffer.data`, but if the implementation changes we don't want to have to alter all the callers.